selfdrive/car/toyota/values.py

```diff
@@ -99,127 +59,278 @@ class ToyotaCarInfo(CarInfo):
   car_parts: CarParts = field(default_factory=CarParts.common([CarHarness.toyota_a]))
 
 
-CAR_INFO: dict[str, ToyotaCarInfo | list[ToyotaCarInfo]] = {
+@dataclass()
+class ToyotaPlatformConfig(PlatformConfig):
+  def init_config(self):
+    if self.platform_str in TSS2_CAR:
+      self.dbc_dict = dbc_dict('toyota_nodsu_pt_generated', 'toyota_tss2_adas')
+    else:
+      if self.platform_str in (CAR.PRIUS, CAR.CHR, CAR.CAMRY, CAR.AVALON_2019):
+        self.dbc_dict = dbc_dict('toyota_nodsu_pt_generated', 'toyota_adas')
+      if self.platform_str in (CAR.AVALON, CAR.HIGHLANDER, CAR.RAV4H, CAR.SIENNA, CAR.LEXUS_IS, CAR.LEXUS_NX, CAR.LEXUS_RC, CAR.LEXUS_RX):
+        self.dbc_dict = dbc_dict('toyota_tnga_k_pt_generated', 'toyota_adas')
+      else:
+        self.dbc_dict = dbc_dict('toyota_new_mc_pt_generated', 'toyota_adas')
```

Comment:
This is the kind of clean up I would hope we'd enable with this refactor (reduce copy pasting), @jnewb1 is this what you had in mind or no?

Reply:
I think these should be the flags that we set, like: https://github.com/commaai/openpilot/pull/31584

(TSS2, NODSU, NO_STOP_TIMER, etc etc)

Reply:
I know we can use flags, but to me grouping seems much cleaner and more maintainable. The logic for group exclusivity is also out in the open

Reply:
well the idea for the refactor was that all the car configuration is within one place, you modify one block of ~10 lines for your car and that's all you have to modify

Reply:
> you could accidentally set multiple exclusive flags

well you can also add a car to two mutually exclusive groups right now, and we can probably setup something to enforce mutually exclusive groups

Reply:
it's a bit of a trade off. Currently, if I want to search for how my corolla is controlled, I would have to search through the whole values.py to see what groups it's a part of, whereas if it's done with flags its more localized

it makes it harder to see which cars are in a group, but I think this is a less common use case